### ISSUE TITLE
use simple_bind can't make continuous predictions

### ISSUE THREAD

**Issue by kyocen** [dmlc/mxnet#4331]
For bugs or installation issues, please provide the following information.
The more information you provide, the more likely people will be able to help you.

## Environment info
Operating System:

**ubuntu 14.04**

Package used (Python/R/Scala/Julia):

**Python**

MXNet version:
installed from source:  

**branch master**

If you are using python package, please provide

**Anaconda 4.2.0 (64-bit)**

Python version and distribution:

**Python 2.7.12**

## Error :

**I train a OCR model use CNN Multi-label net,when I predict a pic one time, the result is correct，but run twice,the first is correct and the second is wrong.**

***

code:

def TestRecognizeOne(file):
       
    ....
    batch_size = 1
    sym, arg_params, aug_params= mx.model.load_checkpoint("ocr", 0)
    data_shape = [("data", (batch_size, 3, 50, 200))]
    input_shapes = dict(data_shape)
    sym = getnet()
    executor = sym.simple_bind(ctx = mx.gpu(), **input_shapes)
    for key in executor.arg_dict.keys():
        if key in arg_params:
            arg_params[key].copyto(executor.arg_dict[key])

    executor.forward(is_train = False, data = mx.nd.array([img]))
    probs = executor.outputs[0].asnumpy()
    
    a=np.argmax(probs[0])
    b=np.argmax(probs[2])
    print a,b

and 

def getnet()
    
    data = mx.symbol.Variable('data')
    label = mx.symbol.Variable('label')
    conv1 = mx.symbol.Convolution(data=data, kernel=(5,5),pad=(2,2), num_filter=96)
    relu1 = mx.symbol.Activation(data=conv1, act_type="relu")
    pool1 = mx.symbol.Pooling(data=relu1, pool_type="max", kernel=(3,3), stride=(2, 2))
    
    conv2 = mx.symbol.Convolution(data=pool1, kernel=(3,3),pad=(1,1), num_filter=128)
    relu2 = mx.symbol.Activation(data=conv2, act_type="relu")
    pool2 = mx.symbol.Pooling(data=relu2, pool_type="max", kernel=(3,3), stride=(2, 2))
    
    flatten = mx.symbol.Flatten(data = pool2)
    fc1 = mx.symbol.FullyConnected(data = flatten, num_hidden = 2048)
    relufc1=mx.symbol.Activation(data=fc1, act_type="relu")
    dropout1 = mx.symbol.Dropout(data=relufc1, p=0.5)

    fc2 = mx.symbol.FullyConnected(data = dropout1, num_hidden = 1024)
    relufc2=mx.symbol.Activation(data=fc2, act_type="relu")
    dropout2 = mx.symbol.Dropout(data=relufc2, p=0.5)

    fc21 = mx.symbol.FullyConnected(data = dropout2, num_hidden = 35)
    fc22 = mx.symbol.FullyConnected(data = dropout2, num_hidden = 35)
    
    fc3 = mx.symbol.Concat(*[fc21, fc22], dim = 0)
    label = mx.symbol.transpose(data = label)
    label = mx.symbol.Reshape(data = label, target_shape = (0, ))
    return mx.symbol.SoftmaxOutput(data = fc3, name = "softmax")
***
when run

 testRecognizeOne("0.jpg")
 testRecognizeOne("0.jpg")

***
the output:
2,3
0,0

the probs:

[ 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0.
0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 1. 0. 0. 0.]

[ 0.02857143 0.02857143 0.02857143 0.02857143 0.02857143 0.02857143
0.02857143 0.02857143 0.02857143 0.02857143 0.02857143 0.02857143
0.02857143 0.02857143 0.02857143 0.02857143 0.02857143 0.02857143
0.02857143 0.02857143 0.02857143 0.02857143 0.02857143 0.02857143
0.02857143 0.02857143 0.02857143 0.02857143 0.02857143 0.02857143
0.02857143 0.02857143 0.02857143 0.02857143 0.02857143]

In addition
when I use simple_bind in lstm-ctc, cnn-ctc, I found the same problem.
